app/cni/pkg/install/main.go

```diff
@@ -124,8 +126,15 @@ func setupChainedPlugin(mountedCniNetDir, cniConfName, kumaCniConfig string) err
 	}
 
 	cniConfPath := path.Join(mountedCniNetDir, resolvedName)
-	if !files.FileExists(cniConfPath) {
-		return errors.Errorf("CNI config %v not found. Kuma CNI won't be chained", cniConfPath)
+	backoff := retry.WithMaxDuration(5*time.Minute, retry.NewConstant(time.Second))
+	err := retry.Do(context.Background(), backoff, func(ctx context.Context) error {
+		if !files.FileExists(cniConfPath) {
+			return errors.Errorf("CNI config %v not found. Kuma CNI won't be chained", cniConfPath)
```

Comment:
I think you have to wrap this with `retry.RetryableError(...)`

Reply:
Yes seems like this is right @StuAtKong